spec/categorical_spec.rb

```diff
@@ -0,0 +1,102 @@
+describe Daru::Vector do
+  context "initialize" do
+    let(:dv) { Daru::Vector.new [:a, 1, :a, 1, :c], type: :category }
+    subject { dv }
+    
+    it { is_expected.to be_a Daru::Vector }
+    its(:size) { is_expected.to eq 5 }
+    its(:type) { is_expected.to eq :category }
```

Comment:
@lokeshh it's `:numeric` not `:number`.

I think pandas dtype reflects the numpy dtype that the data is stored as. daru has `:numeric` and `:object` only so that math and statistics methods can figure out if a computation is feasible on data.

I didn't want to reflect dtype as `int` or `float` since we use `Array` internally.